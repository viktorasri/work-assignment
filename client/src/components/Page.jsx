import React from 'react';
import PropTypes from 'prop-types';
import { Helmet } from 'react-helmet';

const Page = ({ children, title, description }) => {
  return (
    <>
      <Helmet>
        <title>{`${title ? title + ' | ' : ''}Server Explorer`}</title>
        <meta name='description' content={description} />
      </Helmet>
      <main className='page'>{children}</main>
    </>
  );
};

Page.defaultProps = {
  description: 'Server explorer web app',
};

Page.propTypes = {
  description: PropTypes.string,
  title: PropTypes.string,
};

export default Page;
