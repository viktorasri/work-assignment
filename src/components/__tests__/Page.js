import React from 'react';
import { shallow } from 'enzyme';
import Page from '../Page';

describe('<Page />', () => {
  it('expect to render component', () => {
    expect(shallow(<Page />)).toMatchSnapshot();
  });

  it('renders page with passed title prop', () => {
    expect(shallow(<Page title='home' />)).toMatchSnapshot();
  });
});
