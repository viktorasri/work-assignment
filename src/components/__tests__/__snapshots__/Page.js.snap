// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Page /> expect to render component 1`] = `
<Fragment>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      Server Explorer
    </title>
    <meta
      content="Server explorer web app"
      name="description"
    />
  </HelmetWrapper>
  <main
    className="page"
  />
</Fragment>
`;

exports[`<Page /> renders page with passed title prop 1`] = `
<Fragment>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      home | Server Explorer
    </title>
    <meta
      content="Server explorer web app"
      name="description"
    />
  </HelmetWrapper>
  <main
    className="page"
  />
</Fragment>
`;
